compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1076,6 +1076,19 @@ pub struct VarDebugInfo<'tcx> {
 // BasicBlock
 
 rustc_index::newtype_index! {
+    /// The unit of the MIR [control-flow graph][CFG].
+    ///
+    /// There is no branching (e.g., `if`s, function calls, etc.) within a basic block, which makes
+    /// it easier to do [data-flow analyses] and optimizations. Basic blocks consist of a series of
```

Comment:
I use the form with the hyphen here since that seems to be the canonical way: https://en.wikipedia.org/wiki/Data-flow_analysis. Perhaps the rustc-dev-guide should be changed?